src/components/CheckEmailStatusPage/CheckEmailStatusPage.js

```diff
@@ -18,51 +19,53 @@ const CheckEmailStatusPage = (props) => {
       },
     },
   } = props;
+
   const isEmail = validateEmail(userEmail);
   const labelNamespace = 'stripes-core.label';
   const notificationText = isEmail
     ? `${labelNamespace}.sent.email`
     : `${labelNamespace}.your.email`;
+  const hiddenUserEmail = isEmail
+    ? hideEmail(userEmail)
+    : null;
 
   return (
-    <Fragment>
-      <div
-        className={styles.wrap}
-        data-test-status-page
-      >
-        <div className={styles.centered}>
-          <OrganizationLogo />
-          <Headline
-            size="xx-large"
-            tag="h1"
-            data-test-h1
-          >
-            <FormattedMessage id={`${labelNamespace}.check.email`} />
-          </Headline>
-          <Headline
-            size="x-large"
-            tag="p"
-            bold={false}
-            faded
-            data-test-p-notification
-          >
-            <FormattedMessage
-              id={notificationText}
-              {...(isEmail && { values: { userEmail } })}
-            />
-          </Headline>
-          <Headline
-            size="x-large"
-            tag="p"
-            bold={false}
-            faded
-            data-test-p-caution
-          >
-            <FormattedMessage id={`${labelNamespace}.caution.email`} />
-          </Headline>
-        </div>
+    <div
+      className={styles.wrap}
+      data-test-status-page
+    >
+      <div className={styles.centered}>
+        <OrganizationLogo />
+        <Headline
+          size="xx-large"
+          tag="h1"
+          data-test-h1
+        >
+          <FormattedMessage id={`${labelNamespace}.check.email`} />
+        </Headline>
+        <Headline
+          size="x-large"
+          tag="p"
+          bold={false}
+          faded
+          data-test-p-notification
+        >
+          <FormattedMessage
+            id={notificationText}
+            {...(isEmail && { values: { hiddenUserEmail } })}
```

Comment:
Please consider avoiding `isEmail` check here as you already did it for the `hiddenUserEmail `.

Reply:
Thank you for pointing that. The thing is that  `isEmail` results in different `id`s, that is why in one case we need to pass the params but in the other - we do not need to do that